src/transformers/models/tapas/modeling_tapas.py

```diff
@@ -1815,7 +1800,7 @@ def _segment_reduce(values, index, segment_reduce_fn, name):
         dim=0,
     )
 
-    output_values = segment_means.view(new_shape.tolist())
+    output_values = segment_means.clone().view(new_shape.tolist())
```

Comment:
Thanks @Bearnardd for working on this!

BTW, could you teach me how you find we should change this line in order to avoid the issue 
```bash
One of the variables needed for gradient computation has been modified by an in-place operation.
```

I was stuck at this error and didn't know how to figure it out 😢 

Thank you!

Reply:
Also, we should perhaps verify TAPAS can still be fine-tuned properly with this PR.

@Bearnardd could you run [this notebook](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/TAPAS/Fine_tuning_TapasForQuestionAnswering_on_SQA.ipynb) with an installation of Transformers from your branch, to see whether this still works fine? In Colab, you can install it like so:

```
!rm -r transformers
!git clone -b fix_tapas_scatter_2 https://github.com/Bearnardd/transformers.git
!cd transformers
!pip install -q ./transformers
```

Any time you make a change to your branch, you need to restart the runtime and run again to make sure the Transformers version is updated.

Reply:
@NielsRogge - Sure, I will take a a look at it :)

Reply:
Hi @ydshieh - if you take a look at the full message error there is a part that says that `RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [torch.FloatTensor [416]], which is output 0 of ScatterReduceBackward0`. When debugging you can print variable and check its gradient function, as an example `tensor([0., 0., 0.,  ..., 0., 0., 0.], grad_fn=<ScatterReduceBackward0>)` for segment_means which matches the name of the function in the error massage. Moreover there is an indication that we are interested in in-place operation which greatly narrows the list of the suspects.

Reply:
> 

@NielsRogge - I have checked out the notebook and it works fine after updating pytorch to 1.13.